src/lib/components/chat/Settings/Personalization/AddMemoryModal.svelte

```diff
@@ -2,21 +2,60 @@
 	import { createEventDispatcher, getContext } from 'svelte';
 
 	import Modal from '$lib/components/common/Modal.svelte';
-	import { addNewMemory } from '$lib/apis/memories';
+	import { addNewMemory, updateMemoryById } from '$lib/apis/memories';
 	import { toast } from 'svelte-sonner';
 
 	const dispatch = createEventDispatcher();
 
 	export let show;
+	export let memory = {};
+
+	let showUpdateBtn = false;
```

Comment:
I'll update the frontend part!

Reply:
Yes Sir! :+1: 
Appreciate that, first time working with svelte